src/stssession/STSSessionController.ts

```diff
@@ -103,28 +118,28 @@ export class STSSessionController {
       this.allowDownload = true;
     }
 
-    // Set http headers
-    var httpHeaders;
-    if (this.stsHeaderName === undefined || this.stsHeaderValue === undefined) {
-      httpHeaders = {
-        headers: {
-          'Content-Type': 'application/json'
-        }
-      };
-    } else {
+    // set extra http headers
+    if (!this.stsHeaderName === undefined && !this.stsHeaderValue === undefined) {
       var name = this.stsHeaderName;
-      httpHeaders = {
+      this.httpHeaders = {
```

Comment:
I made an earlier comment on this but I see you're changing this so you can ignore the previous one.